### ISSUE TITLE
Annotations on Rancher defined Pod Security Policies are not applied on Kubernetes cluster PSP

### ISSUE THREAD

**Issue by gillarda** [rancher/rancher#22093]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create a new PSP via Global > Security > Pod Security Policies > Add Policy.
2. Add annotations to PSP, for example : 
`"seccomp.security.alpha.kubernetes.io/defaultProfileName": "runtime/default"`
3. Apply the new PSP to a project

**Expected Result:**

User specified annotations should be set on PSP on Kubernetes clusters, in order to be enforced.

**Actual Results:**

PSP on Kubernetes clusters lack the annotations.

**Other details that may be helpful:**

**Environment information**
- Rancher version : 2.2.6
- Installation option (single install/HA): HA

**Cluster information**
- Kubernetes version (use `kubectl version`):
```
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:36:19Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

**PSP state in Rancher API**
```
{
  "allowPrivilegeEscalation": false,
  "allowedCapabilities": null,
  "allowedFlexVolumes": null,
  "allowedHostPaths": null,
  "allowedProcMountTypes": null,
  "allowedUnsafeSysctls": null,
  "annotations": {
    "lifecycle.cattle.io/create.cluster-pspt-sync_c-jrxhd": "true",
    "lifecycle.cattle.io/create.cluster-pspt-sync_local": "true",
    "lifecycle.cattle.io/create.mgmt-pspt-lfc-handler": "true",
    "seccomp.security.alpha.kubernetes.io/allowedProfileNames": "runtime/default",
    "seccomp.security.alpha.kubernetes.io/defaultProfileName": "runtime/default"
  },
  "baseType": "podSecurityPolicyTemplate",
  "created": "2019-08-09T08:28:46Z",
  "createdTS": 1565339326000,
  "creatorId": "user-kj7gd",
  "defaultAddCapabilities": null,
  "defaultAllowPrivilegeEscalation": false,
  "description": null,
  "forbiddenSysctls": null,
  "id": "restricted-enhanced",
  "labels": {
    "cattle.io/creator": "norman"
  },
  "name": "restricted-enhanced",
  "privileged": false,
  "readOnlyRootFilesystem": false,
  "andOther": "stuff"
}
```
**PSP state on Kubernetes cluster**
```
apiVersion: extensions/v1beta1
kind: PodSecurityPolicy
metadata:
  annotations:
    serviceaccount.cluster.cattle.io/pod-security: restricted-enhanced
    serviceaccount.cluster.cattle.io/pod-security-version: "206055832"
  creationTimestamp: "2019-08-09T10:48:09Z"
  labels:
    cattle.io/creator: norman
  name: restricted-enhanced-psp
  resourceVersion: "2630352"
  selfLink: /apis/extensions/v1beta1/podsecuritypolicies/restricted-enhanced-psp
  uid: 2d49dbff-ba93-11e9-a9ee-0236dadab8ea
spec:
  allowPrivilegeEscalation: false
  defaultAllowPrivilegeEscalation: false
  requiredDropCapabilities:
  - ALL
  andOther: stuff

```

gz#8109

gz#8711

gz#9416

gz#14323

**Comment by kinarashah** [comment]
As mentioned in gz, we only pick selected annotations: https://github.com/rancher/rancher/blob/master/pkg/controllers/managementuser/rbac/podsecuritypolicy/template.go#L83

**Comment by rawmind0** [comment]
Submitted PR https://github.com/rancher/rancher/pull/31651 to provide this feature. 

The feature will add annotations defined at Rancher psp template to applied downstream cluster psp. To avoid potencial beside effects, just annotations with a name that  doesn't contain `.cattle.io/` will be applied.

**Comment by kravciak** [comment]
I tried to reproduce on v2.5-head (fa087c8), but there is no option to apply PSP on project level if it's not enabled for cluster first.


To reproduce I did:
```
 - go to Global -> Security -> PSP -> Add Policy
 - fill in the name, create policy with all the defaults (can't add anotations yet)
 - edit the created policy, add annotation: 
   seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default

 - create 1-node infrastructure provider cluster (DigitalOcean)
 - in cluster advanced options select "Enable" PSP Support , select PSP I created as default policy
```

Display PSP in rancher api 
```
~ Global -> Security -> PSP -> thepolicy -> view in API
...
  "annotations": {
    "cleanup.cattle.io/psptUpgrade": "true",
    "lifecycle.cattle.io/create.mgmt-pspt-lfc-handler": "true",
    "seccomp.security.alpha.kubernetes.io/defaultProfileName": "runtime/default"
  },
...
```

Display PSP annotations in kubernetes cluster
```
~ kubectl get psp thepolicy-psp -o json | jq '.metadata.annotations'
{
  "serviceaccount.cluster.cattle.io/pod-security": "thepolicy",
  "serviceaccount.cluster.cattle.io/pod-security-version": "18832"
}
```

**Comment by kravciak** [comment]
I tested fix on 2.5-head (b2a34d7) and master-head (440b4e5), it works fine.

```
~ kubectl get psp thepolicy-psp -o json | jq '.metadata.annotations'
{
  "seccomp.security.alpha.kubernetes.io/defaultProfileName": "runtime/default",
  "serviceaccount.cluster.cattle.io/pod-security": "thepolicy",
  "serviceaccount.cluster.cattle.io/pod-security-version": "3348
}
```
